### ISSUE TITLE
Automate deployment of iOS framework for Tangram

### ISSUE THREAD

**Issue by karimnaaji** [tangrams/ios-framework#5]
From https://github.com/tangrams/tangram-es/pull/1003 Tangram Framework can now be built multiple ways:
- `TangramMap.framework` for architectures `armv7s`, `armv7` and `arm64` (~30mb).
- `TangramMap.framework` for architectures `i386`, `x86_64` (~25mb).
- Universal `TangramMap.framework` for architectures all of the architectures (`armv7s`, `armv7`, `arm64`, `i386`, `x86_64`) (~50mb).

Framework sizes are subject to change. Those can be either built for `Debug` or `Release`.

Considering all those options we should decide on what to publish in this repository. 

It seems that we don't have to code sign the framework (@msmollin please confirm):

> there is no need for framework Developer to distribute their framework to other developers with codesigning identity in it because Consumer will anyway re-codesign it.

But:

>  iOS frameworks containing simulator slices can't be submitted to the App Store

From http://stackoverflow.com/questions/30963294/creating-ios-osx-frameworks-is-it-necessary-to-codesign-them-before-distributin#31126203 

Our options seems to be either to:
- publish one uncodesigned Framework for `Release` for all of the architectures, and give proper instructions to developers of how to remove simulator slices within their build system
- publish two uncodesigned Frameworks, one configured for either `Debug` or `Release` containing device + simulator architectures (for dev) and one for `Release` containing only device architectures (for publishing).

cc @msmollin @blair1618 @ecgreb

**Comment by msmollin** [comment]
There is one additional option here: direct people to integrate this using Cocoapods or Carthage (should make a card to include Carthage support in the future).

My Thoughts:
We should heavily lean towards people integrating using one of the existing dependency management solutions because they handle this issue. Cocoapods automatically generates a shell script and updates your project to run it with strips out whichever framework archs are invalid for the build step. Carthage has instructions for a similar script since Carthage is a more hands-off solution to begin with (see the link in the above SO question).

Mapbox also leans heavily towards using Cocoapods for similar reasons, and Google only provides instructions for using Cocoapods, so this is not without precedent. 

For people looking to integrate this manually, we can direct them towards https://github.com/realm/realm-cocoa/blob/master/scripts/strip-frameworks.sh as an example for them as to how to remove the invalid architectures at build time. They'll have to write their own script.

**Comment by msmollin** [comment]
However I would follow the lead of all the other major providers a provide a single universal framework that end developers/consumers are required to codesign.

**Comment by karimnaaji** [comment]
Thanks for the details! It seems that cocoapods does all of what is required for us, so this works for the iOS SDK.

We could still give some instructions about stripping architectures (or give a shell script snippet to add to the developer build system), so all of our developers are not required to use cocoapods or carthage.

**Comment by oursnoir** [comment]
Hi !
I'm looking into this repository for some times, I wonder if the cocoapod integration is working (especially regarding the simulator support) since the README.md doesn't seem to reflect the actual advancement of the project.

Best

**Comment by msmollin** [comment]
@oursnoir Thanks for the question! Cocoapod integration is indeed working, and you are very correct that we haven't upgraded the README to indicate what the status is of the project.

We still haven't finished exposing all the features from the core codebase (or even come equivalent with the Android SDK) so the plan is to update the readme once all that is in place. However I have opened an issue (https://github.com/tangrams/ios-framework/issues/11) to make sure it doesn't get forgotten about. Once we've finished development for the 1.0 release, we'll be submitting this up to Cocoapods central spec repo! 

To see how we're using it currently inside the Mapzen SDK, take a look over here: https://github.com/mapzen/ios/blob/master/Podfile

and

https://github.com/mapzen/ios/blob/master/ios-sdk/TangramVC.swift

Thanks for the question! In the future, feel free to open a new issue so we don't muddy the waters on this one too much!

**Comment by msmollin** [comment]
To be clear though: simulator support has been integrated and works well.

**Comment by oursnoir** [comment]
Ouah ! I'm really glad to hear all of that !
I will definitely take a look to all the links you mentioned.

Best

**Comment by msmollin** [comment]
This work has been completed as of 0.7 release. In the event there are issues we should open tickets in the main ES repo.
